features/lorawan/lorastack/phy/LoRaPHYAS923.h

```diff
@@ -0,0 +1,363 @@
+     *                +---+---+---+---+---+---+---+---+
+     *                            |
+     *                            +---+---+---+---+---+---+---+---+
+     *Downlink preamble 8 symbols |   |   |   |   |   |   |   |   |
+     *                            +---+---+---+---+---+---+---+---+
+     *
+     *                     Worst case Rx window timings
+     *
+     * TRxLate  = DEFAULT_MIN_RX_SYMBOLS * tSymbol - RADIO_WAKEUP_TIME
+     * TRxEarly = 8 - DEFAULT_MIN_RX_SYMBOLS * tSymbol - RxWindowTimeout - RADIO_WAKEUP_TIME
+     *
+     * TRxLate - TRxEarly = 2 * DEFAULT_SYSTEM_MAX_RX_ERROR
+     *
+     * RxOffset = ( TRxLate + TRxEarly ) / 2
+     *
+     * RxWindowTimeout = ( 2 * DEFAULT_MIN_RX_SYMBOLS - 8 ) * tSymbol + 2 * DEFAULT_SYSTEM_MAX_RX_ERROR
+     * RxOffset = 4 * tSymbol - RxWindowTimeout / 2 - RADIO_WAKE_UP_TIME
+     *
+     * The minimum value of RxWindowTimeout must be 5 symbols which implies that the system always tolerates at least an error of 1.5 * tSymbol
+     */
+    /*!
+     * Computes the RX window timeout and offset.
+     *
+     * \param [in] datarate     The RX window datarate index to be used.
+     *
+     * \param [in] minRxSymbols The minimum required number of symbols to detect an RX frame.
+     *
+     * \param [in] rxError      The system maximum timing error of the receiver in milliseconds.
+     *                          The receiver will turn on in a [-rxError : +rxError] ms
+     *                          interval around RxOffset.
+     *
+     * \param [out] rxConfigParams The updated WindowTimeout and WindowOffset fields.
+     */
+    virtual void compute_rx_win_params(int8_t datarate,
+                                                 uint8_t minRxSymbols,
+                                                 uint32_t rxError,
+                                                 RxConfigParams_t *rxConfigParams);
+
+    /*!
+     * \brief TX configuration.
+     *
+     * \param [in] txConfig A pointer to the function parameters.
+     *
+     * \param [out] txPower The TX power index set.
+     *
+     * \param [out] txTimeOnAir The time-on-air of the frame.
+     *
+     * \retval True, if the configuration was applied successfully.
+     */
+    virtual bool tx_config(TxConfigParams_t* txConfig, int8_t* txPower,
+                                TimerTime_t* txTimeOnAir );
+
+    /*!
+     * \brief The function processes a Link ADR Request.
+     *
+     * \param [in] linkAdrReq A pointer to the function parameters.
+     *
+     * \param [out] drOut The datarate applied.
+     *
+     * \param [out] txPowOut The TX power applied.
+     *
+     * \param [out] nbRepOut The number of repetitions to apply.
+     *
+     * \param [out] nbBytesParsed The number of bytes parsed.
+     *
+     * \retval The status of the operation, according to the LoRaMAC specification.
+     */
+    virtual uint8_t link_ADR_request(LinkAdrReqParams_t* linkAdrReq,
```

Comment:
`int8_t* nbBytesParsed );` some have a space before `)` , some does also after `(`, some do not have any